exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -324,6 +324,11 @@ protected boolean setupNewSchema() throws SchemaChangeException {
               if (name == EMPTY_STRING) {
                 continue;
               }
+
```

Comment:
In current implementation dirN columns will be returned in select all clause during querying directory as before.